samples/publish_datasource.py

```diff
@@ -0,0 +1,81 @@
+####
+# This script demonstrates how to use the Tableau Server Client
+# to publish a datasource to a Tableau server. It will publish
+# a specified datasource to the 'default' project of the provided site.
+#
+# Some optional arguments are provided to demonstrate async publishing,
+# as well as providing connection credentials when publishing. If the
+# provided datasource file is over 64MB in size, TSC will automatically
+# publish the datasource using the chunking method.
+#
+# For more information, refer to the documentations:
+# (https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_ref_datasources.htm#publish_data_source)
+#
+# To run the script, you must have installed Python 3.5 or later.
+####
+
+import argparse
+import getpass
+import logging
+
+import tableauserverclient as TSC
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Publish a datasource to server.')
+    parser.add_argument('--server', '-s', required=True, help='server address')
+    parser.add_argument('--site', '-i', help='site name')
+    parser.add_argument('--username', '-u', required=True, help='username to sign into server')
+    parser.add_argument('--filepath', '-f', required=True, help='filepath to the datasource to publish')
+    parser.add_argument('--logging-level', '-l', choices=['debug', 'info', 'error'], default='error',
+                        help='desired logging level (set to error by default)')
+    parser.add_argument('--as-job', '-a', help='Publishing asynchronously', action='store_true')
```

Comment:
We should just call this "--async" because "as job" doesn't communicate what is happening, imo.

Reply:
Yeah, I agree with this. Fixed in new commit